### PR TITLE
Add max_num_validation_steps member of config and client and related …

### PR DESCRIPTION
**Pull request by jewelltaylor**
# PR Type
[**Feature** | Fix | Documentation | Other ]

# Short Description

[Add ability to specify maximum number of validation steps to perform](https://app.clickup.com/t/868bcpawz)

Add ability to specify maximum number of validation steps to perform. This functionality is useful in cases where the validation dataset size is large and we don't want to compute metrics over the entire validation set. This is immediately useful for the nnunet experiments because the [nnunet cli](https://github.com/MIC-DKFZ/nnUNet/blob/43349fa5f0680e8109a78dca7215c19e258c9dd7/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py#L151) only performs 50 steps on the validation set.

# Tests Added

Describe the tests that have been added to ensure the codes correctness, if applicable.

### CODE REVIEW

File: fl4health/clients/basic_client.py [hunk 1]
```diff
@@ -748,7 +753,10 @@ def _validate_or_test(
         metric_manager.clear()
         loss_meter.clear()
         with torch.no_grad():
-            for input, target in maybe_progress_bar(loader, self.progress_bar):
+            for i, (input, target) in enumerate(maybe_progress_bar(loader, self.progress_bar)):
```
Comment (scarere) [inline fl4health/clients/basic_client.py:756]: So this approach works great for the nnunet client where the dataloaders randomly sample from the set. But if we have a traditional dataloader that runs through the val set, then will setting a max number of steps here that doesn't cover the whole validation set mean that we're basically not using part of the validation set? If so do we care? Is it worth trying to add logic that ensures the whole validation set is used?

**Review by scarere — APPROVED** [review]

File: fl4health/clients/basic_client.py (hunk 1 above)
Comment (jewelltaylor) [reply fl4health/clients/basic_client.py:756]: I think it's a lever you would only want to turn on if you explicitly don't want to use the entire validation set (ex: for performance reasons). In all other instances, you would not specify max_num_validation_steps and get the expected behaviour of going through entire validation set

Comment (jewelltaylor) [reply fl4health/clients/basic_client.py:756]: I will add some sort of log that warns that we are sampling from validation set when max_validation_steps is specified
